File: app/src/main/java/com/thesunnahrevival/sunnahassistant/views/TimePickerFragment.java
package com.thesunnahrevival.sunnahassistant.views;

import android.app.Dialog;
import android.app.TimePickerDialog;
import android.os.Bundle;
import android.text.format.DateFormat;
import android.util.Log;
import android.widget.TimePicker;

import java.util.Calendar;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;
import androidx.lifecycle.MutableLiveData;

public class TimePickerFragment extends DialogFragment
        implements TimePickerDialog.OnTimeSetListener {

    public static final String TAG = "TimePickerFragment";
    static MutableLiveData<String> timeSet = new MutableLiveData<>();

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        // Use the current time as the default values for the picker
        final Calendar c = Calendar.getInstance();
        int hour = c.get(Calendar.HOUR_OF_DAY);
        int minute = c.get(Calendar.MINUTE);
        if (timeSet.getValue() != null) {
            try {
                String[] strings = timeSet.getValue().split(":");
                hour = Integer.parseInt(strings[0]);
                minute = Integer.parseInt(strings[1]);
            } catch (NumberFormatException e) {
                Log.v(TAG, "String cannot be converted to  valid time");
            }

        }


        // Create a new instance of TimePickerDialog and return it
        return new TimePickerDialog(getActivity(), this, hour, minute,
                DateFormat.is24HourFormat(getActivity()));
    }

    public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
        StringBuilder time = new StringBuilder();
        if (hourOfDay < 10)
            time.append("0");
        time.append(hourOfDay);
        time.append(":");
        if (minute < 10)
            time.append("0");
        time.append(minute);
        timeSet.setValue(time.toString());
    }
}

File: app/src/main/java/com/thesunnahrevival/sunnahassistant/views/dialogs/TimePickerFragment.kt
package com.thesunnahrevival.sunnahassistant.views.dialogs

import android.app.Dialog
import android.app.TimePickerDialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.os.Bundle
import android.text.format.DateFormat
import android.util.Log
import android.widget.TimePicker
import androidx.fragment.app.DialogFragment
import androidx.lifecycle.MutableLiveData
import com.thesunnahrevival.sunnahassistant.utilities.formatTimeInMilliseconds
import com.thesunnahrevival.sunnahassistant.utilities.getTimestampInSeconds
import java.util.*

class TimePickerFragment : DialogFragment(), OnTimeSetListener {
    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        // Use the current time as the default values for the picker
        val c = Calendar.getInstance()
        var hour = c[Calendar.HOUR_OF_DAY]
        var minute = c[Calendar.MINUTE]
        if (timeSet.value != null) {
            try {
                timeSet.value?.let {
                    val strings =  it.split(":").toTypedArray()
                    hour = strings[0].toInt()
                    minute = strings[1].toInt()
                }
            } catch (e: NumberFormatException) {
                Log.v(TAG, "String cannot be converted to  valid time")
            }
        }


        // Create a new instance of TimePickerDialog and return it
        return TimePickerDialog(activity, this, hour, minute,
                DateFormat.is24HourFormat(activity))
    }

    override fun onTimeSet(view: TimePicker, hourOfDay: Int, minute: Int) {
        val hour: String
        val amOrPm: String
        if (view.is24HourView) {
            hour = hourOfDay.toString()
            amOrPm = ""

        } else if (!view.is24HourView && hourOfDay < 12) {
            hour = hourOfDay.toString()
            amOrPm = " am"
        } else {
            hour = (hourOfDay - 12).toString()
            amOrPm = " pm"
        }


        val time = StringBuilder()
        if (hourOfDay < 10) {
            time.append("0")
        }
        time.append(hour)
        time.append(":")
        if (minute < 10)
            time.append("0")
        time.append(minute)
        time.append(amOrPm)
        val timeInMilliseconds = getTimestampInSeconds(time.toString()) * 1000
        timeSet.value = formatTimeInMilliseconds(requireContext(), timeInMilliseconds)
    }

    companion object {
        const val TAG = "TimePickerFragment"
        var timeSet = MutableLiveData<String?>(null)
    }
}